src/power_grid_model/validation/validation.py

```diff
@@ -331,7 +379,25 @@ def validate_required_values(
         required["line"] += ["r0", "x0", "c0", "tan0"]
         required["shunt"] += ["g0", "b0"]
 
-    return list(chain(*(none_missing(data, component, required.get(component, [])) for component in data)))
+    # Mark `power_sigma` for each power_sensor based on the state of `p_sigma` and `q_sigma`
+    process_power_sigma_and_p_q_sigma(data, "sym_power_sensor", required)
+    process_power_sigma_and_p_q_sigma(data, "asym_power_sensor", required)
+
```

Comment:
i'm still wondering why the alternative implementation of making all three of `power_sigma`, `p_sigma` and `q_sigma` optional and then just doing a sanity check such that

```py
s_nan = np.nan(power_sigma)
p_nan = np.nan(p_sigma)
q_nan = np.nan(q_sigma)
if np.any(np.logical_xor(p_nan, q_nan)):
    raise ValidationError("Only one p_sigma and q_sigma is provided")
if np.any(np.logical_and(s_nan, p_nan)):
    raise ValidationError("Neither power_sigma nor p_sigma and q_sigma is provided")
```

Reply:
This approach would mean that for every future such check we would have to write a specific function. While if we make the `required` field tailored to the data, which we are doing anyways, we might as well just do it the right way. Wouldn't you agree?

Reply:
`required` really means required without any changes. if you look at the implementation for the trio (`sn`, `uk`, `pk`), then there are additional requirements imposed to the explicit structure.

in this specific case, neither `power_sigma` nor `p_sigma` nor `q_sigma` are `required` in the classical sense (since if you only look at `power_sigma`, then it is technically `optional`). Instead, there's an additional requirement that either `power_sigma` and/or both `p_sigma` and `q_sigma` need to be provided. It may raise the same exception as the `required` checks, but it should not follow the same code path

Reply:
> required really means required without any changes.

Are you referring to "changes to the `required` field" or "changes to the data"?

> in this specific case, neither power_sigma nor p_sigma nor q_sigma are required in the classical sense

`power_sigma` most definitely have lived in `required::*sym_power_sensor` for quite a while and not `optional`.

In short, either the previous logic is problematic or misleading OR it's not the way you personally understood/liked it.

Reply:
That is because `power_sigma` was required previously. Now it is not, as long as `p_sigma` and `q_sigma` were provided